src/3d/qgscameracontroller.cpp

```diff
@@ -307,6 +307,22 @@ void QgsCameraController::onPositionChanged( Qt3DInput::QMouseEvent *mouse )
   }
 }
 
+void QgsCameraController::screenPointToWorldPos( QPoint position, Qt3DRender::QCamera *cameraBefore,
+    std::function<void( double, const QVector3D & )> onSuccess )
+{
+  double depth = sampleDepthBuffer( mDepthBufferImage, position.x(), position.y() );
+  if ( std::isnan( depth ) )
+    qDebug() << "screenPointToWorldPos: depth is NaN.";
```

Comment:
I don't know if it is possible, but could it be also infinite?

```suggestion
  if ( std::isfinite( depth ) )
    qDebug() << "screenPointToWorldPos: depth is NaN or Inf.";
```